dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -280,6 +280,43 @@ def new_csf_pre201
       render_csf_survey(PRE_DEEPDIVE_SURVEY, workshop)
     end
 
+    # Display CSF101 (Intro) post-workshop survey.
+    # The survey, on submit, will display thanks.
+    # GET workshop_survey/csf/post101(/:enrollment_code)
+    def new_csf_post101
+      # Use enrollment_code to find a specific workshop
+      # or search all CSF101 workshops the current user is enrolled in.
+      enrolled_workshops = nil
+      if params[:enrollment_code].present?
+        enrolled_workshops = Workshop.joins(:enrollments).
+          where(pd_enrollments: {code: params[:enrollment_code]})
```

Comment:
does this check if the workshop already happened/is that a requirement (I would guess it is since it's a post survey?)

Reply:
Yes, we check for attendance [just below](https://github.com/code-dot-org/code-dot-org/pull/33272/files#diff-f3d236167f6f73de2351be6d507f492fR305-R309).